docs/Phoenix Walkthrough.md

```diff
@@ -0,0 +1,331 @@
+simple Phoenix application from scratch and take it through 4
+releases. 1 main release, and 3 hot upgrade releases.
+
+**NOTE** At this time this guide does not cover
+[Ecto](https://github.com/elixir-ecto/ecto)'s use in releases. This
+will be added at a later time.
+
+### First Steps
+
+First off we will create a new Phoenix app (without Ecto) using `mix
+phoenix.new --no-ecto phoenix-distillery`. Go ahead and fetch
+dependencies when prompted by the mix task.
+
+Don't forget to run an `npm install` from within your project
+directory in order to install brunch and its dependencies!
+
+Next we will install distillery in our `mix.exs`
+
+*file: mix.exs*
+```elixir
+  defp deps do
+    [ ...,
+     {:distillery, "~> 0.9"},
+      ...,
+    ]
+  end
+```
+
+Execute a `mix deps.get, compile` and you are ready to continue.
+
+### Distillery Configuration
+
+To initialize Distillery, run `mix release.init`. Please refer to the
+Distillery walkthrough for a detailed look at the configuration
+options available.
+
+We will need to configure the `prod` environment before we start
+building releases.
+
+*file: config/prod.exs*
+```elixir
+config :phoenix_distillery, PhoenixDistillery.Endpoint,
+  http: [port: {:system, "PORT"}],
+  url: [host: "localhost", port: {:system, "PORT"}], # This is critical for ensuring web-sockets properly authorize.
+  cache_static_manifest: "priv/static/manifest.json",
+  server: true,
+  root: ".",
+  version: Mix.Project.config[:version]
+```
+
+Let's discuss each of these options.
+
+- `server` configures the endpoint to boot the
+  [Cowboy](https://github.com/ninenines/cowboy) application http
+  endpoint on start.
+- `root` configures the application root for serving static files
+- `version` ensures that the asset cache will be busted on *versioned*
+  application upgrades (more on this later)
+
+**NOTE** We are telling our release to use ENV variables by providing
+the tuple `{:system, "PORT"}` to the port option. Your release will
+not start properly if the `PORT` variable is not available to it on
+the production machine/in the production environment.
+
+
+### Version 0.0.1
+
+Before we modify our app at all, we will generate a release with the
+current state. We will generate a release. Our release command is 3
+separate commands which I have linked with `&&`:
+
+`./node_modules/brunch/bin/brunch b -p && MIX_ENV=prod mix phoenix.digest && mix release --env=prod`
+
+1. `./node_modules/brunch/bin/brunch b -p` builds your assets in
+   production mode. More detail can be found in the
+   [Phoenix Static Asset Guide](http://www.phoenixframework.org/docs/static-assets)
+1. `MIX_ENV=prod mix phoenix.digest` To compress and tag your assets
+    for proper caching. More detail can be found in the
+    [Phoenix Mix Task Guide](http://www.phoenixframework.org/docs/mix-tasks#section--mix-phoenix-digest-)
+1. `mix release --env=prod` To actually generate a release for a
```

Comment:
I would add `MIX_ENV=prod` here as well, so it's clear we're compiling for prod *and* release the prod release.